### DISCUSSION TITLE
isDirty is not working correctly after reset

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#3687] [General]
After `reset({ name: "My Name" })`

1. Type "a" on the "name" input => "My Namea"
2. `isDirty` become true
3. Remove "a" => "My Name"
4. `isDirty` still is true (Expected: `isDirty` become false)

https://codesandbox.io/s/react-hook-form-get-started-forked-wfqq5?file=/src/index.js:668-694

**Comment by bluebill1049** [comment]
provide a codesandbox plz

**Reply by kaisiant** [reply]
> provide a codesandbox plz

Ops, I forgot to attach it.

**Reply by bluebill1049** [reply]
you should reset the other two inputs value as well inside the `reset`. it's better that way, because in hook form we don't assume what's in the inputs rather trust you what you have supplied, if you are lazy you can use getValues() to combine with your other inputs which got reseted.

**Reply by kaisiant** [reply]
How should I register "id" and reset the "id" value coz isDirty is still true.
https://codesandbox.io/s/react-hook-form-get-started-forked-wfqq5?file=/src/index.js:747-787

**Reply by bluebill1049** [reply]
i don't see the id in the defautlValues?

**Reply by kaisiant** [reply]
Ya, I am using the form in create and update UI. I am not sure is it necessary to include the id in defaultValues but what I can confirm is  need to register('id') in order to include the id into the form values

**Reply by bluebill1049** [reply]
`isDirty` is basically checked again `defaultValues` with your input.
